packages/@o3r/schematics/src/utility/update-pipes.ts

```diff
@@ -0,0 +1,115 @@
+  fileWithImports: string,
+  templateFile: string,
+  tree: Tree,
+  context: SchematicContext
+) => {
+  if (ts.isObjectLiteralExpression(ngDecorator.expression.arguments[0])) {
+    const importsProp = ngDecorator.expression.arguments[0].properties.find((prop): prop is ts.PropertyAssignment & { initializer: ts.ArrayLiteralExpression } =>
+      ts.isPropertyAssignment(prop)
+              && ts.isIdentifier(prop.name)
+              && prop.name.text === 'imports'
+              && ts.isArrayLiteralExpression(prop.initializer)
+    );
+    if (importsProp) {
+      matchers.forEach((matcher) => {
+        const pipeName = matcher[1];
+        const importModuleRegexp = new RegExp(`\\b${pipeReplacementInfo[pipeName].import}\\b`, 'g');
+        if (importsProp.initializer.elements.some((element) => importModuleRegexp.test(element.getText()))) {
+          if (pipeReplacementInfo[pipeName].new.import) {
+            tree.overwrite(
+              fileWithImports,
+              tree.readText(fileWithImports).replaceAll(importModuleRegexp, pipeReplacementInfo[pipeName].new.import!)
+            );
+          }
+          tree.overwrite(
+            templateFile,
+            tree.readText(templateFile).replaceAll(new RegExp(`\\|\\s*${pipeName}`, 'g'), `| ${pipeReplacementInfo[pipeName].new.name}`)
+          );
+        } else {
+          context.logger.warn(`
+            Deprecated pipe ${pipeName} was found in ${templateFile},
+            but no associated declaration in ${fileWithImports} to be able to migrate to the new pipe ${pipeReplacementInfo[pipeName].new.name}.
+          `);
+        }
+      });
+    }
+  }
+};
+
+const isNgModuleDecorator = (decorator: ts.Decorator) =>
+  ts.isCallExpression(decorator.expression)
+  && decorator.expression.expression
+  && ts.isIdentifier(decorator.expression.expression)
+  && decorator.expression.expression.text === 'NgModule';
+
+/**
+ * Returns a rule tu update pipes
+ * @param pipeReplacementInfo
+ */
+export const updatePipes = (pipeReplacementInfo: PipeReplacementInfo): Rule => (tree, context) => {
+  const pipeRegex = new RegExp(`\\|\\s*(${Object.keys(pipeReplacementInfo).join('|')})`, 'g');
+  const files = findFilesInTree(tree.root, (file) => file.endsWith('.html'));
+  files.forEach((file) => {
+    const matchers = Array.from(
+      file.content.toString().matchAll(pipeRegex)
+    ).filter((matcher) => !!pipeReplacementInfo[matcher[1]]);
+    if (matchers.length) {
+      const directory = dirname(file.path);
+      const baseFileName = basename(file.path, '.template.html');
+      const componentFile = join(directory, `${baseFileName}.component.ts`);
+      const moduleFile = join(directory, `${baseFileName}.module.ts`);
+      let standalone = false;
+      if (tree.exists(componentFile)) {
+        try {
+          const info = getO3rComponentInfoOrThrowIfNotFound(tree, componentFile);
+          standalone = info.standalone;
+        } catch {}
+      }
```

Comment:
can be valuable to add a `else` case with a info message that we have detected the usage of pipes in this template but, because we didn't find the module, the update will be skipped.
Something like:
```typescript
logger.info(`Pipe renaming will be skipped for "${file.path}" because the associated module has not been detected.`);
```